### ISSUE TITLE
[Feature Request] move pandas to extras

### ISSUE THREAD

**Issue by psarka** [DLR-RM/stable-baselines3#680]
### 🚀 Suggestion

Move pandas dependency to extras.

### Motivation

Pandas is a heavy dependency, doing very little for stable baselines and quite annoying when running on a device such as Nvidia Jetson, as I need to build a wheel for it myself. I may find time to contribute this movement, if you would be interested.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

**Comment by Miffyli** [comment]
Good point and I agree with this (heavy dependency for little functionality). @araffin what is your view on this? We could (potentially?) replace all panda functionality with other dependency-free code, unless that breaks some backwards-compatibility stuff.

**Comment by psarka** [comment]
I inspected the code, so let me make a more precise suggestion:

`pandas` is used only in three places:
- `monitor/load_results`
- `logger/read_json`
- `logger/read_csv`

Of the three the last two functions are only used in tests, so they could be moved there. `load_results` ir only used in `results_plotter`.

`matplotlib`, another dependency that I would like not to have, is only used in two places:
- `logger/Figure`
- `results_plotter`

in `logger/Figure` it is only used as a type annotation, so easy to do without. 

That leaves the `results_plotter` as the only user of both pandas and matplotlib. So my updated proposal would be to:

- move `read_json` to tests
- move `read_csv` to tests
- move `load_results` to `results_plotter`
- move `pandas` and `matplotlib` to optional dependencies in the `extra` section, with the impact that `results_plotter` will no longer work with base installation. Logger will still work.
- if needed, add `pandas` to optional dependencies `tests` section.

**Comment by araffin** [comment]
Hello,
rather than changing the install behavior, I would be more in favor of making SB3 run without `pandas` and `matplotlib`, so when you install it with `pip install --no-deps` flag.
That would mean doing lazy import of `pandas` inside `load_results` (and other helpers) and using `if typing.TYPE_CHECKING` in the logger.
